torch/csrc/jit/codegen/cuda/index_reference_replay.cpp

```diff
@@ -170,10 +170,14 @@ TensorDomain* IndexReferenceReplay::computeReplay() {
       continue;
     }
 
+    // Make a copy of the root_id for the reference to "own"
+    IterDomain* root_id_copy = new IterDomain(
```

Comment:
Is it ok to leave iter_type_ & ir_rfactor_domain_ to its default value without copying them from root_id?

Reply:
So I tried to use `root_id->clone()` instead and it actually triggers the old error in the BNRepro2
`Merging IterDomains requires that their iteration types match.`
This makes me nervous and I'm not sure if that means we are just hiding some legit errors with this code change. Since apparently ownership is not the root cause of the error.

But I don't have the expertise to dig it deeper, I'll update the comment and merge this patch as-is.
Meanwhile, cc'ing @naoyam @csarofeen  so we keep this in the backlog.

Reply:
@jjsjann123 Thanks for checking. That's indeed concerning. I'll take a look later.

Reply:
@jjsjann123  that was on purpose. We want the standard IterType, and we want false for rfactor on this.

Reply:
Christian's right. Here, we just want to have a plain IterDomain. We probably don't need to set the parallel type as well.

Reply:
Sounds good. I'll merge if after upstream master pull